plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php

```diff
@@ -13,6 +13,12 @@
  */
 class Tax extends Task {
 
+	/**
+	 * Used to cache is_complete() method result.
+	 * @var null
+	 */
+	private $is_complete_result = null;
```

Comment:
A static cache is a step forward. However, an actual cache entry set via `wp_cache_set()` would be preferable. We can invalidate the cache item(s) by implementing the action hook [`woocommerce_after_(object_type)_object_save`](https://wp-kama.com/plugin/woocommerce/hook/woocommerce_after_(object_type)_object_save) (for object_type "product").

Reply:
Thanks, @sun 

I think you meant to leave the comment for Products.php, not Tax.php.

With that said, I agree that caching the result is better. However, [wp_count_posts](https://developer.wordpress.org/reference/functions/wp_count_posts/) already caches using `wp_cache_set`.